packages/services/src/kernel/future.ts

```diff
@@ -140,7 +140,12 @@ class KernelFutureHandler extends DisposableDelegate implements Kernel.IFuture {
     this._iopub = Private.noOp;
     this._reply = Private.noOp;
     if (!this._testFlag(Private.KernelFutureFlag.IsDone)) {
+      // Reject the `done` promise, but catch its error here in case no one else
+      // is waiting for the promise to resolve. This prevents the error from
+      // being displayed in the console but does not prevent it from being
+      // caught by a client who is waiting for it.
       this._done.reject(new Error('Canceled'));
+      this._done.promise.catch(reason => { /* no-op */ });
```

Comment:
I think I've solved this in my PR, though I do it by waiting for the done promise, rather than allowing the cancellation to actually happen. I like this idea, though. Can we put this one line in my PR, and save the rest of the changes for after the PR? I'm happy to put it in, or you can push to my branch.

The yarn.lock changes should probably already be in master, with the webpack upgrade? Also, a lot of other changes may already be done in my PR.